### PR TITLE
Update to use the latest crates

### PR DESCRIPTION
**Pull request by m5p3nc3r**
display-interface -> 0.5.0
embedded-hal -> 1.0.0
smart-leds -> 0.4.0
smart-leds-trait -> 0.3.0

This enables me to integrate my driver with the latests smart-leds implementation

### CODE REVIEW

**Comment by libesz** [comment]
Thanks! I noticed some incompatibility with smart-leds-trait 0.3.0. Added a commit that fixes it. Could you please verify?

File: src/lib.rs [hunk 1]
```diff
@@ -59,9 +59,10 @@ where
         self.writer.write(iter)
     }
     pub fn flush_with_gamma(&mut self) -> Result<(), T::Error> {
-        let iter = brightness(gamma(
-            self.content.as_slice().iter().cloned()
-        ), self.brightness);
+        let iter = brightness(
```
Comment (m5p3nc3r) [inline src/lib.rs:62]: I believe this is only a whitespace change?

File: src/lib.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use embedded_graphics_core::{
     Pixel,
 };
 
-use smart_leds::{brightness, gamma, RGB8, SmartLedsWrite};
+use smart_leds::{brightness, gamma, SmartLedsWrite, RGB8};
```
Comment (m5p3nc3r) [inline src/lib.rs:17]: This is just an ordering change

File: src/lib.rs [hunk 3]
```diff
@@ -114,7 +115,7 @@ mod tests {
 
         fn write<T, I>(&mut self, iterator: T) -> Result<(), Self::Error>
         where
-            T: Iterator<Item = I>,
+            T: IntoIterator<Item = I>,
```
Comment (m5p3nc3r) [inline src/lib.rs:118]: Yup - this fixes a warning.  I had this locally, not sure why it didn't make it into the MR.

File: src/lib.rs [hunk 4]
```diff
@@ -126,7 +127,7 @@ mod tests {
         }
     }
 
-    fn get64pixels(color: Rgb888) -> ([Pixel<Rgb888>; 64]) {
+    fn get64pixels(color: Rgb888) -> [Pixel<Rgb888>; 64] {
```
Comment (m5p3nc3r) [inline src/lib.rs:130]: This is fine, it removes a linting error, but no functional change

File: src/lib.rs (hunk 1 above)
Comment (m5p3nc3r) [reply src/lib.rs:62]: Hey - I have reviewed the changes.  Thanks for making them.

I believe the first two are just whitespace or ordering changes.  No functional change
The third is valid, I had this locally, but it wasn't in the MR - my bad, sorry!
The fourth is also valid but not a functional change.  It fixes a linting error.

**Review by libesz — APPROVED** [review]

**Comment by libesz** [comment]
Looks good. Will release a new version with this PR merged.
Thanks!

**Comment by m5p3nc3r** [comment]
Awesome, thanks for the quick turnaround on this.

**Pull request merged**